stackslib/src/chainstate/burn/db/sortdb.rs

```diff
@@ -1830,6 +1842,39 @@ impl<'a> SortitionHandleConn<'a> {
         SortitionHandleConn::open_reader(connection, &sn.sortition_id)
     }
 
+    /// Does the sortition db expect to receive unknown blocks from
+    /// this tenure?
+    ///
+    /// This is used by nakamoto nodes while they are at or near the
+    /// current chain tip: only recent tenures can receive blocks this
+    /// way. Otherwise, the `BlockHeaderHash` must have been
+    /// explicitly confirmed by a block commit.
+    pub fn expects_blocks_from_tenure(
```

Comment:
I'm having a hard time reconciling the docstring with the code body.  The code body returns the _latest_ `BlockSnapshot` whose block-signing key hash matches the given public key, if there any such snapshots within the last `NAKAMOTO_TENURE_BLOCK_ACCEPTANCE_PERIOD` sortitions.  What does this have to do with "expect[ing] to receive unknown blocks?"  Also, it seems miners can reuse the same signing key in multiple tenures, so if there were two snapshots from the same miner in this window, this method would not return the earlier one.  Is this desired?

Reply:
Good point. I think this should be a bool instead.

At first, I intended to use this result to figure out the consensus hash that started the miner's tenure. However, for exactly the reason you point out, it cannot do that (because the miner's key could be reused). Instead, I think that the Nakamoto block headers should contain the consensus hash of their tenure. There's other ways to figure out their associated tenure, I'm sure, but it dramatically simplifies things if the block's include it. I know it's a somewhat counter-intuitive change from Stacks 2.0's perspective: in 2.0, a miner cannot know their consensus hash when they produce their block, but because block production happens _after_ their tenure starts in 3.0, they actually **do** know their consensus hash.

Reply:
> Instead, I think that the Nakamoto block headers should contain the consensus hash of their tenure.

I agree -- we should use this instead of `BurnchainHeaderHash` for `burn_view`.

Reply:
I'll do this when working on the coordinator